### PR TITLE
Improve check_assessment_access tests

### PR DESCRIPTION
**Pull request by trombonekenny**
This PR started as the following:

> This PR changes the way the CBTF scheduler enforces reservation-to-PL-assessment matching. It removes the date range matching (and fuzzy PS to PL matching), and exclusively uses the `exam_uuid` matching that's been in place for a few semesters now.
>
> Courses that do not use the scheduler for their "mode": "Exam" exams can disable the reservation checking by setting course_instance.ps_linked to false. (ZJUI)

However, the removal of PrairieSchedule code was done in other PRs, so this PR now just adds some additional tests for the `check_assessment_access` sproc.

### CODE REVIEW

**Comment by nwalters512** [comment]
We've since moved away from using PrairieSchedule in favor of PrairieTest; closing this.

**Comment by trombonekenny** [comment]
Actually this one is still relevant. We need to remove the access rules that expect PS linking. This comes up with the ZJUI mode=Exam situations.

**Comment by trombonekenny** [comment]
But it's old code that probably needs review/refactoring.

**Comment by nwalters512** [comment]
@trombonekenny gotcha, sorry for closing prematurely. Can you resolve the conflicts and get this ready for review?

**Comment by trombonekenny** [comment]
Closes #1373 (not sure why that wasn't linking here, sorry the names are a bit out of whack)

**Comment by trombonekenny** [comment]
@mwest1066 The re-written sproc tests are passing, but `access.test.js` where it did things with PrairieSchedule integration. Should we remove that and rely only on the sproc testing? Or update access.test.js to do the PT test and logic?

**Comment by trombonekenny** [comment]
I went ahead and removed the PS tests. I'll leave it to @mwest1066 and his review (poke) to decide if he'd like to add PT tests here. Those could also be added under a different PR.

**Comment by trombonekenny** [comment]
@mwest1066 *poke* I updated this to work with the new PL repo structure but it still needs review.

File: apps/prairielearn/src/sprocs/check_assessment_access_rule.sql [hunk 1]
```diff
@@ -45,103 +43,48 @@ BEGIN
...
+            AND e.user_id = check_assessment_access_rule.user_id;
+
+        IF FOUND THEN
+            IF assessment_access_rule.exam_uuid = exam_uuid THEN
+                -- exam_uuid matches, so don't keep going to authorized := FALSE
                 EXIT schedule_access;
+            ELSE
+                -- checked-in so deny any exams that is not linked
+                authorized := FALSE;
             END IF;
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/sprocs/check_assessment_access_rule.sql:98]: I just tested, and it is possible for a student to have more than one active reservation at a time, so I assume the exam_uuid will need to go to the query instead of to a separate test.

**Comment by trombonekenny** [comment]
Notes from a dev meeting today:

- This PR refactors some things that are PrairieTest related. Avoid touching that.
- Move ahead with stripping out PrairieScheduler code and tests.
- Get tests to pass against master

Take out things in one PR to the level that it passes.
Make another PR for any refactoring, if necessary.

File: apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts [hunk 2]
```diff
@@ -1,59 +1,225 @@
...
+      //     authz_mode: 'Exam',
+      //     course_role: 'None',
+      //     course_instance_role: 'None',
+      //     user_id: '1000',
+      //     uid: 'valid@school.edu',
+      //     date: '2010-07-07 06:06:06-00',
+      //     display_timezone: 'US/Central',
+      //   });
+      //   assert.isFalse(authorized);
+      // });
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts:195]: @trombonekenny from what I can tell, this test tries to assert behavior that doesn't currently exist. Let's discuss. I don't think we necessarily want the behavior that this test expects, as for better or for worse we still support legacy `exam_mode_networks` independently of PT and `exam_uuid`.

Comment (trombonekenny) [reply apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts]: Is there an edge case where someone is in mode: Exam and there's an exam with mode: Exam protections but no exam_uuid?

Comment (nwalters512) [reply apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts]: I wouldn't call it an edge case, but yes, that's what would happen if someone is using `exam_mode_networks`, which some self-hosters are.

Comment (nwalters512) [reply apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts]: Or are you thinking about the case where a student is in a PT center and they end up with access to one exam with an `examUuid` and one without?

Comment (trombonekenny) [reply apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts]: Yeah, I was looking for tests for all the possible permutations of access like mode: Exam, checked in reservation, exam protected by uuid (and their not counterparts.)

Comment (nwalters512) [reply apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts]: I agree that it's useful to have assertions in place; the question is what behavior we actually want to assert. Exam-mode assessments without UUIDs is indeed a thing we support, so I think we should assert that access is allowed in that case.

Comment (mwest1066) [reply apps/prairielearn/src/tests/sprocs/check_assessment_access.test.ts]: At the dev meeting today we concluded that we do care about the case when we are in exam mode but don't have an examUuid. However, to handle this case correctly we'll need to remember whether we are in exam mode because of PT or because of exam mode networks. Passing this information down to the assessment access checks will require passing down an extra variable.

**Review by mwest1066 — APPROVED** [review]
